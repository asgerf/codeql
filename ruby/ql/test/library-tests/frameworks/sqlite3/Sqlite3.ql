private import codeql.ruby.DataFlow
private import codeql.ruby.Concepts
private import codeql.ruby.frameworks.Sqlite3
private import codeql.ruby.security.SqlInjectionCustomizations

query predicate sqlite3SqlConstruction(SqlConstruction c, DataFlow::Node sql) { sql = c.getSql() }

query predicate sqlite3SqlExecution(SqlExecution e, DataFlow::Node sql) { sql = e.getSql() }

query predicate sqlinjectionSinks(SqlInjection::Sink sink) { any() }
