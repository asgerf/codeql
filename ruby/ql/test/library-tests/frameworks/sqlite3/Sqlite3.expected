sqlite3SqlConstruction
sqlite3SqlExecution
sqlinjectionSinks
| sqlite3.rb:5:12:5:17 | <<-SQL |
| sqlite3.rb:12:19:12:41 | "select * from numbers" |
| sqlite3.rb:17:15:17:35 | "select * from table" |
| sqlite3.rb:29:19:29:39 | "select * from table" |
